### PR TITLE
Calendar

### PR DESCRIPTION
**Pull request by marcusvolz**
Added a calendar heatmap using the calmap package. Will look at tweaking the aesthetics (e.g. color) later.

### CODE REVIEW

**Comment by marcusvolz** [comment]
Checks have failed - will look into it later.

**Comment by hugovk** [comment]
```
Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\hostedtoolcache\windows\Python\3.7.9\x64\Scripts\stravavis.exe\__main__.py", line 7, in <module>
  File "C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\site-packages\stravavis\cli.py", line 37, in main
    from stravavis.plot_calendar import plot_calendar
  File "C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\site-packages\stravavis\plot_calendar.py", line 1, in <module>
    import calmap
ModuleNotFoundError: No module named 'calmap'
Error: Process completed with exit code 1.
```

Add `calmap` to `setup.cfg` so it also gets installed:

```cfg
install_requires =
    calmap
    fit2gpx
    gpxpy
    matplotlib
    pandas
    rich
    seaborn
```

File: src/stravavis/cli.py [hunk 1]
```diff
@@ -28,15 +34,20 @@ def main():
...
     from stravavis.plot_landscape import plot_landscape
     from stravavis.plot_map import plot_map
+    from stravavis.process_activities import process_activities
     from stravavis.process_data import process_data
 
     print("Processing data...")
     df = process_data(args.path)
 
+    print("Processing activities...")
+    activities = process_activities(activities_path)
```
Comment (hugovk) [inline src/stravavis/cli.py]: There's no `activities_path` here, we need to pass `args.activities_path`:

```suggestion
    activities = process_activities(args.activities_path)
```

File: src/stravavis/plot_calendar.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    activities['date'] = activities['Activity Date'].dt.date
+    activities = activities.groupby(['date'])['Distance'].sum()
+    activities.index = pd.to_datetime(activities.index)
+    activities.clip(0, max_dist, inplace=True)
+    
+    if year_min is not None:
+        activities = activities[activities.index.year>=year_min]
+    
+    if year_max is not None:
+        activities = activities[activities.index.year<=year_max]
```
Comment (hugovk) [inline src/stravavis/plot_calendar.py]: We can simplify this a bit.

We skip when these are `None`, and we can skip if they're falsy (`False`, 0, `None`).

```suggestion
    if year_min:
        activities = activities[activities.index.year>=year_min]
    
    if year_max:
        activities = activities[activities.index.year<=year_max]
```

**Review by hugovk — COMMENTED** [review]
Something else, `--activities_path` is an optional parameter with no default so this would fail:

```console
$ stravavis tests/gpx
Processing data...
Processing ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Processing activities...
Traceback (most recent call last):
...
ValueError: Invalid file path or buffer object type: <class 'NoneType'>
```

Some options:

1. make it a required, positional argument, like `"path"`. But maybe people want to run on GPX only and have no `activities.csv`.

2. Add a default or calculate one. Current directory? Same as `args.path`? Hard to know.

3. My preference: if the argument is given, skip the relevant bits. For example:

```diff
diff --git a/src/stravavis/cli.py b/src/stravavis/cli.py
index c84dc2c..f7fd3f0 100644
--- a/src/stravavis/cli.py
+++ b/src/stravavis/cli.py
@@ -45,8 +45,10 @@ def main():
     print("Processing data...")
     df = process_data(args.path)
 
-    print("Processing activities...")
-    activities = process_activities(activities_path)
+    activities = None
+    if args.activities_path:
+        print("Processing activities...")
+        activities = process_activities(args.activities_path)
 
     print("Plotting facets...")
     outfile = f"{args.output_prefix}-facets.png"
@@ -68,10 +70,11 @@ def main():
     plot_landscape(df, output_file=outfile)
     print(f"Saved to {outfile}")
 
-    print("Plotting calendar...")
-    outfile = f"{args.output_prefix}-calendar.png"
-    plot_calendar(activities, output_file=outfile)
-    print(f"Saved to {outfile}")
+    if activities:
+        print("Plotting calendar...")
+        outfile = f"{args.output_prefix}-calendar.png"
+        plot_calendar(activities, output_file=outfile)
+        print(f"Saved to {outfile}")
 
 
 if __name__ == "__main__":
```

**Comment by marcusvolz** [comment]
Many thanks Hugo, if I missed anything let me know!

**Pull request merged**